examples/secret_channel.py

```diff
@@ -0,0 +1,106 @@
+# import discord and any other libraries
+import discord
+from discord.ext import commands
+from discord.utils import get
+
+# bot description
+description = "A simple example showcasing how to make text and voice channels (as well as private ones)."
+
+# create a bot instance with prefix "?" and description
+bot = commands.Bot(command_prefix="?", description=description)
+
+@bot.event
+async def on_ready():
+    """Prints information to the console once the bot is ready."""
+
+    print('Logged in as')
+    print(bot.user.name)
+    print(bot.user.id)
+    print('Bot Online!')
+
+def can_create():
+    """A check to see if the user has the required manage_channels permission."""
+    
+    async def predicate(ctx):
+        member = ctx.author
+        return member.guild_permissions.manage_channels
+    return commands.check(predicate)
+
+@bot.group(invoke_without_command=True)
+@can_create()
+async def create(ctx, *, new_channel: str):
+    """Command to create an open text channel."""
+
+    # the passed argument new_channel is used to name the created channel
+    channel = await ctx.guild.create_text_channel(new_channel, overwrites=None, category=None, reason=None)
+    await channel.send(f"Text channel {new_channel} was created!") # sends a message into that channel
+
+@create.command()
+@can_create()
+async def voice(ctx, *, new_channel: str):
+    """Subcommand to create an open voice channel."""
+
+    # note: ctx.guild.categories will return a list of possible categories that the server has
+    # you can then select where the channel will be created in the category list
+    channel = await ctx.guild.create_voice_channel(new_channel, category=ctx.guild.categories[1])
```

Comment:
This code will fail if a server has no categories. Likewise, it's a bit odd to just fetch the second one in the list out of the blue.